.github/workflows/cost-estimate.yml

```diff
@@ -0,0 +1,335 @@
+          with open('cost-estimate-output/summary-table.md', 'w') as f:
+              f.write(table)
+              
+          print("Cost parsing completed")
+          EOF
+          
+          python3 parse_costs.py
+
+      # === Generate Summary ===
+      - name: Generate cost summary
+        if: always()
+        run: |
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "## 📊 Cost Estimate Summary" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          
+          # Add the parsed cost table
+          if [ -f "cost-estimate-output/summary-table.md" ]; then
+            cat cost-estimate-output/summary-table.md >> $GITHUB_STEP_SUMMARY
+          else
+            echo "Cost summary table could not be generated." >> $GITHUB_STEP_SUMMARY
+          fi
+          
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "### Test Details" >> $GITHUB_STEP_SUMMARY
+          echo "- **Toolchain**: ${{ matrix.toolchain }}" >> $GITHUB_STEP_SUMMARY
+          echo "- **Test Exit Code**: ${{ steps.cost-test.outputs.test_exit_code }}" >> $GITHUB_STEP_SUMMARY
+          echo "- **Snapshot Drift**: ${{ steps.cost-test.outputs.snapshot_drift }}" >> $GITHUB_STEP_SUMMARY
+          echo "- **Commit SHA**: ${{ github.sha }}" >> $GITHUB_STEP_SUMMARY
+          
+          # Add link to full logs
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "📄 **Full test logs are available in the workflow artifacts.**" >> $GITHUB_STEP_SUMMARY
+
+      # === Upload Artifacts ===
+      - name: Upload cost estimate logs
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: cost-estimate-logs-${{ matrix.toolchain }}
+          path: |
+            cost-estimate-output/
+          retention-days: 30
+
+      # === Check for Failures ===
+      - name: Check test results
+        if: always()
+        run: |
+          # Fail the job if tests failed
+          if [ "${{ steps.cost-test.outputs.test_exit_code }}" != "0" ]; then
+            echo "Cost estimate tests failed"
+            exit 1
+          fi
+          
+          # Fail the job if snapshot drift detected
+          if [ "${{ steps.cost-test.outputs.snapshot_drift }}" == "true" ]; then
+            echo "Expect-test snapshot drift detected"
+            echo "The cost estimates have changed. Please review and update snapshots if the changes are expected."
+            exit 1
+          fi
+          
+          echo "All cost estimate tests passed"
+
+  # === Aggregate Results (runs after matrix jobs complete) ===
+  cost-estimate-summary:
+    name: Cost Estimate Summary
+    runs-on: ubuntu-latest
+    needs: cost-estimate
+    if: always()
+    
+    steps:
+      - name: Download all artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: cost-estimate-logs-*
+          merge-multiple: true
+
+      - name: Create consolidated summary
+        run: |
```

Comment:
Not sure we need any of this as we have GITHUB_STEP_SUMMARY and we don't want to have a test matrix here.